src/imageMagickDiff.js

```diff
@@ -0,0 +1,171 @@
+import child_process from 'child_process';
+import path from 'path';
+
+var DimensionOfImage;
+
+//TODO: for large images, perhaps split it into smaller pieces and then compare to optimize
+var doDiff  =  async function(filename1, filename2, diffFile, includeOnlyBoxes, skipBoxes) {
+
+  var colourOfDiff = '"rgba(255, 0, 255, 255)"' ; // inner double quotes necessary
+
+  var paddedImages = await resizeImages(filename1, filename2);
+  var paddedImage1 = paddedImages[0];
+  var paddedImage2 = paddedImages[1];
+
+  if(!!includeOnlyBoxes && includeOnlyBoxes.length) {
+    await includeOnlyAsync([paddedImage1, paddedImage2], includeOnlyBoxes);
+  }
+
+  if(!!skipBoxes && skipBoxes.length) {
+    await maskOutAsync([paddedImage1, paddedImage2], skipBoxes);
+  }
+
+  return new Promise(function (resolve) {
+
+    diffFile = diffFile || 'miff:';
+
+    var compare_process = child_process.exec('compare -metric AE' + ' ' + '-highlight-color'  + ' ' + colourOfDiff + ' '
+        + paddedImage1 + ' ' + paddedImage2 + ' ' + diffFile  + ' && '
+        + 'rm' + ' ' + paddedImage1 + ' ' + paddedImage2);
+
+    compare_process.on('exit', function (code) {
+      if(code === 0) {
+        resolve(false);
+      } else {
+        resolve(true);
+      }
+    })
+  });
+};
+
+var extensionAddedName = function (extension, filePath) {
+  return path.join(path.dirname(filePath), extension + path.basename(filePath));
+};
+
+var resizeImages = async function(filename1, filename2) {
+
+  var paddedImage1 = extensionAddedName('padded+', filename1);
+  var paddedImage2 = extensionAddedName('padded+', filename2);
+
+  var [height1, height2, width1, width2] = (await Promise.all([
+    checkStdOut('convert' + ' ' + filename1 + ' ' + '-ping -format "%h" info:'),
```

Comment:
Wasn't this supposed to be chained into a single command?

Reply:
chaining into one command would be possible but would have to parse the resulting string and this would reduce speed overall